serde-generate/runtime/golang/src/lcs/deserializer.go

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) Facebook, Inc. and its affiliates
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+package lcs
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/facebookincubator/serde-reflection/serde"
+)
+
+// MaxSequenceLength is max length allowed for sequence
+const MaxSequenceLength = (1 << 31) - 1
+
+const maxUint32 = uint64(^uint32(0))
+
+// Deserializer implements `serde.Deserializer` interface for deserializing LCS serialized bytes
+type Deserializer struct {
+	buf *bytes.Buffer
+}
+
+// NewDeserializer creates a new `serde.Deserializer`
+func NewDeserializer(input []byte) serde.Deserializer {
+	return &Deserializer{buf: bytes.NewBuffer(input)}
+}
+
+func (d *Deserializer) DeserializeBytes() ([]byte, error) {
+	len, err := d.DeserializeLen()
+	if err != nil {
+		return nil, err
+	}
+	ret := make([]byte, len)
+	_, err = d.buf.Read(ret)
+	return ret, err
+}
+
+func (d *Deserializer) DeserializeLen() (int, error) {
+	ret, err := d.deserializeUleb128AsU32()
+	if ret > MaxSequenceLength {
+		return 0, errors.New("length is too large")
+	}
+	return int(ret), err
+}
+
+func (d *Deserializer) DeserializeStr() (string, error) {
+	bytes, err := d.DeserializeBytes()
+	return string(bytes), err
+}
+
+func (d *Deserializer) DeserializeBool() (bool, error) {
+	ret, err := d.buf.ReadByte()
+	return ret == 1, err
+}
+
+func (d *Deserializer) DeserializeUnit() (struct{}, error) {
+	return struct{}{}, nil
+}
+
+// DeserializeChar is unimplemented
+func (d *Deserializer) DeserializeChar() (rune, error) {
+	panic("unimplemented")
```

Comment:
Perhaps better throw an error than panicking, since we can.

Reply:
this is for keep it consistent with SortMapEntries, which does not have error returned.